operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/event/source/informer/TemporaryResourceCache.java

```diff
@@ -84,14 +103,25 @@ public synchronized void putResource(T newResource, String previousResourceVersi
     var cachedResource = getResourceFromCache(resourceId)
         .orElse(managedInformerEventSource.get(resourceId).orElse(null));
 
-    if ((previousResourceVersion == null && cachedResource == null)
+    boolean moveAhead = false;
+    if (previousResourceVersion == null && cachedResource == null) {
+      if (tombstones.get(newResource.getMetadata().getUid()) != null) {
+        log.debug(
+            "Won't resurrect uid {} for resource id: {}",
+            newResource.getMetadata().getUid(), resourceId);
+        return;
+      }
+      moveAhead = true;
```

Comment:
THis seems to be correct, but would be good to document why is this here, isn't is covered already by the condition on line 120-121 ? or just an optimization?

Reply:
I'll add a comment, it's just breaking apart the checks that were being performed.

Reply:
thank you!